testgrid/conformance/upload_e2e.py

```diff
@@ -0,0 +1,206 @@
+#
+# Usage: see README.md
+
+
+import re
+import sys
+import time
+from datetime import datetime
+import argparse
+import json
+import subprocess
+from os import path
+
+# NOTE e2e logs use go's time.StampMilli ("Jan _2 15:04:05.000")
+# Example log line with a timestamp:
+# Jan 26 06:38:46.284: INFO: Running AfterSuite actions on all node
+# the third ':' separates the date from the rest
+E2E_LOG_TIMESTAMP_RE = re.compile('(... .\d \d\d:\d\d:\d\d\.\d\d\d):.*')
+
+# Ginkgo gives a line like the following at the end of successful runs:
+# SUCCESS! -- 123 Passed | 0 Failed | 0 Pending | 587 Skipped PASS
+# we match this to detect overall success
+E2E_LOG_SUCCESS_RE = re.compile('SUCCESS! -- .* PASS')
+
+def parse_e2e_log_line_timestamp(line, year):
+    """parses a ginkgo e2e log line for the leading timestamp
+
+    Args:
+        line (str) - the log line
+        year (str) - 'YYYY'
+
+    Returns:
+        timestamp (datetime.datetime) or None
+    """
+    match = E2E_LOG_TIMESTAMP_RE.match(line)
+    if match is None:
+        return None
+    # note we add year to the timestamp because the actual timestamp doesn't
+    # contain one and we want a datetime object...
+    timestamp = year+' '+match.group(1)
+    return datetime.strptime(timestamp, '%Y %b %d %H:%M:%S.%f')
+
+def parse_e2e_logfile(f, year):
+    """parse e2e logfile at path, assuming the log is from year
+    
+    Args:
+        f (file): the log file, iterated for lines
+        year (str): YYYY year logfile is from
+
+    Returns:
+        started (datetime.datetime), finished (datetime.datetime), passed (boolean)
+    """
+    started = finished = None
+    passed = False
+    for line in f:
+        # try to get a timestamp from each line, keep the first one as
+        # start time, and the last one as finish time
+        ts = parse_e2e_log_line_timestamp(line, year)
+        if started is None:
+            started = ts
+        elif ts:
+            finished = ts
+        # if we found the ginkgo success line then the run passed
+        is_success = E2E_LOG_SUCCESS_RE.match(line)
+        if is_success:
+            passed = True
+    return started, finished, passed
+
+def datetime_to_unix(dt):
+    """convert datetime.datetime to unix timestamp"""
+    return int(time.mktime(dt.timetuple()))
+
+def testgrid_started_json_contents(start_time):
+    """returns the string contents of a testgrid started.json file
+
+    Args:
+        start_time (datetime.datetime)
+
+    Returns:
+        contents (str)
+    """
+    started = datetime_to_unix(start_time)
+    return json.dumps({
+        "timestamp": started
+    })
+
+def testgrid_finished_json_contents(finish_time, passed):
+    """returns the string contents of a testgrid finished.json file
+
+    Args:
+        finish_time (datetime.datetime)
+        passed (bool)
+
+    Returns:
+        contents (str)
+    """
+    finished = datetime_to_unix(finish_time)
+    result = "SUCCESS" if passed else "FAILURE"
```

Comment:
Please use `'` or `"` consistently within the file  `'SUCCESS'`

Reply:
done